net: lib: Fix MDNS RESPONDER

loicpoulain opened the pull request:
1. Fix malformed MDNS response

2. The dns directory was not included when building with
DNS_RESONDER or LLMR_RESPONDER configs.

Commit:
net: lib: Fix missing dns subdir include

The dns directory was not included when building with
MDNS_RESONDER or LLMR_RESPONDER configs.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

tbursztyka commented on subsys/net/lib/CMakeLists.txt:
```diff
@@ -5,12 +5,17 @@ add_subdirectory_if_kconfig(coap)
 add_subdirectory_ifdef(CONFIG_TLS_CREDENTIALS   tls_credentials)
 
+if (CONFIG_DNS_RESOLVER
+    OR CONFIG_DNS_RESPONDER
```
According to the Kconfig, you cannot have CONFIG_DNS_RESPONDER nor CONFIG_LLMNR_RESPONDER without CONFIG_DNS_RESOLVER

What was the config you were trying to apply?

jukkar reviewed (commented):
According to Kconfig file this should not be needed.
The Kconfig file has
```
config DNS_RESOLVER
	bool "DNS resolver"

if DNS_RESOLVER

config MDNS_RESOLVER
	bool "Enable mDNS support"

config LLMNR_RESOLVER
	bool "Enable LLMNR support"

endif
```

I just wonder should we put `depends on DNS_RESOLVER` to mDNS and LLMNR configs.

loicpoulain commented:
@tbursztyka , I'm building MDNS sample which has CONFIG_MDNS_RESPONDER (only). 
@jukkar, it's **MDNS_RESPONDER** not RESOLVER, so DNS_RESOLVER should not be a dependency.

tbursztyka commented:
@loicpoulain So Kconfig should be fixed as well, not only the cmake file.

loicpoulain commented:
> @loicpoulain So Kconfig should be fixed as well, not only the cmake file.

What do you mean? in Kconfig, MDNS_RESPONDER, is correctly defined out of DNS_RESOLVER if block.

Commit:
net: mdns_resolver: Fix malformed MDNS response

The query->len does not take 'Terminator' into account, shift over
one byte to prevent overwriting it with next field (Type).

This fixes mdns_resolver sample.

Fixes: 87eb552dd27 (net/dns: Switch mdns responder to new net_pkt API)

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

loicpoulain commented:
ping
